### ISSUE TITLE
implicit cast of uint/int in equality ops riddle

### ISSUE THREAD

**Issue by onelson** [influxdata/flux#5069]
In https://github.com/influxdata/flux/pull/5042 a bug in the `>=` and `>` operators for `int`/`uint` and `uint`/`int` arguments prompted a "reality check" discussion.

The the riddle before us is: should `-1 > uint(v: 1)` be true or false?

In a situation where we wrap the (signed) int `-1` in a call to `uint()`, we can see the value "wraps around":

```
> uint(v: -1)
18446744073709551615
```

With this, a `uint` cast of `-1` actually could be considered larger than `uint` `1`.

The question: should `uint` on either side force the cast of _the other side_ to `uint`, or should `int` on either side force the cast to `int`?

Consider the precedent set by `float`. When either side is a `float`, the _other side_ is cast to `float` before comparing. To me this means we should be selecting the weaker/more permissive type in these situations. A "race to the bottom."

The implementation we have today avoids casting at all by reasoning that a negative `int` value _must be smaller than even the smallest `uint`_, and conversely that a negative `int` _cannot be larger than any valid `uint`_.

---

In summary this might be a #wontfix. We might not want to change a thing, but we had a brief discussion on slack about it so it seemed worth recording here for posterity.

**Comment by onelson** [comment]
While this issue was mostly concerned with implicit casting during equality ops _in general_, there's a related concern about `uint` (where these casts are the most questionable) behaving differently between x86 and arm architectures. See: #5274
